core/capabilities/ccip/ccipsolana/msghasher.go

```diff
@@ -0,0 +1,113 @@
+package ccipsolana
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/utils/ccip"
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/utils/tokens"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// MessageHasherV1 implements the MessageHasher interface.
+// Compatible with:
+// - "OnRamp 1.6.0-dev"
+type MessageHasherV1 struct {
+	lggr logger.Logger
+}
+
+func NewMessageHasherV1(lggr logger.Logger) *MessageHasherV1 {
+	return &MessageHasherV1{
+		lggr: lggr,
+	}
+}
+
+// Hash implements the MessageHasher interface.
+func (h *MessageHasherV1) Hash(_ context.Context, msg cciptypes.Message) (cciptypes.Bytes32, error) {
+	h.lggr.Debugw("hashing message", "msg", msg)
+
+	anyToSolanaMessage := ccip_router.Any2SVMRampMessage{}
+	anyToSolanaMessage.Header = ccip_router.RampMessageHeader{
+		SourceChainSelector: uint64(msg.Header.SourceChainSelector),
+		DestChainSelector:   uint64(msg.Header.DestChainSelector),
+		SequenceNumber:      uint64(msg.Header.SequenceNumber),
+		MessageId:           msg.Header.MessageID,
+		Nonce:               msg.Header.Nonce,
+	}
+	anyToSolanaMessage.TokenReceiver = solana.PublicKeyFromBytes(msg.Receiver)
+	anyToSolanaMessage.Sender = msg.Sender
+	anyToSolanaMessage.Data = msg.Data
+	for _, ta := range msg.TokenAmounts {
+		anyToSolanaMessage.TokenAmounts = append(anyToSolanaMessage.TokenAmounts, ccip_router.Any2SVMTokenTransfer{
+			SourcePoolAddress: ta.SourcePoolAddress,
+			DestTokenAddress:  solana.PublicKeyFromBytes(ta.DestTokenAddress),
+			ExtraData:         ta.ExtraData,
+			Amount:            ccip_router.CrossChainAmount{LeBytes: tokens.ToLittleEndianU256(ta.Amount.Int.Uint64())},
```

Comment:
We also need to set DestGasAmount, derived and encoded from DestExecDataDecoded here.
Previously I didn't add it because it wasn't yet added to the hasher onchain.
The field should be encoded with borsch encoding